app/containers/LanguageProvider/reducer.js

```diff
@@ -0,0 +1,26 @@
+/*
+ *
+ * LanguageProvider reducer
+ *
+ */
+
+import { fromJS } from 'immutable';
+import {
+  CHANGE_LOCALE,
+} from './constants';
+
+const initialState = fromJS({
+  locale: 'en',
```

Comment:
maybe we should try to grab browser locale? ie

```js
window.navigator.userLanguage || window.navigator.language;
```

if we do try to detect user locale, we will need to check it against our supported languages (`appLocales` export from `i18n`)